packages/gateway/scripts/denylist.js

```diff
@@ -0,0 +1,149 @@
+import fetch from '@web-std/fetch'
+import path from 'path'
+import toml from 'toml'
+import { CID } from 'multiformats'
+import { base32 } from 'multiformats/bases/base32'
+import { sha256 } from 'multiformats/hashes/sha2'
+import fs from 'fs'
+import * as uint8arrays from 'uint8arrays'
+
+/**
+ * @typedef {{ id: string, title: string }} Namespace
+ * @typedef {{ name: string, metadata: any }} Key
+ * @typedef {{ key: string, value: any, metadata?: any }} BulkWritePair
+ */
+
+const rootDir = path.dirname(path.dirname(import.meta.url))
+const wranglerConfigPath = `${rootDir}/wrangler.toml`
+const denyListPath = `${rootDir}/denylist.json`
+
+const DENY_LIST_SOURCES = [
+  'https://badbits.dwebops.pub/denylist.json',
+  denyListPath,
+]
+
+export async function denylistSyncCmd({ env }) {
```

Comment:
Should we trigger this in a daily cron job ?

Reply:
Yes, but I'll open an issue to get that done...